src/__experimental__/components/select/select.component.js

```diff
@@ -188,88 +170,163 @@ class Select extends React.Component {
+      formattedValue: this.formattedValue(this.state.filter, value)
+    };
+
+    return props;
+  }
+
+  listDisplayable(isTypeAhead) {
+    const { filter, open } = this.state;
+    return (this.openWhenTypeAhead((isTypeAhead), filter) || !isTypeAhead) ? open : false;
   }
 
   render() {
     const {
       children,
-      className,
       customFilter,
       placeholder,
       value,
       onLazyLoad,
       onFilter,
       onOpen,
+      typeAhead,
+      filterable,
       ...props
     } = this.props;
 
-    let events = {};
-
-    if (!this.props.disabled && !this.props.readOnly) {
-      events = {
-        onBlur: this.handleBlur,
-        onChange: this.handleFilter,
-        onFocus: this.handleFocus,
-        onKeyDown: this.handleKeyDown,
-        onClick: this.handleClick
-      };
-    }
+    const { filter, open } = this.state;
 
+    const allowTypeAhead = filterable ? typeAhead : false;
     return (
       <>
-        <InputDecoratorBridge
+        <Textbox
           { ...props } // this needs to send all of the original props
-          data-component='carbon-select'
-          className={ this.className(className) }
-          formattedValue={ this.formattedValue(this.state.filter, value) }
-          inputIcon={ this.isMultiValue(value) ? undefined : 'dropdown' }
-          inputRef={ this.assignInput }
-          placeholder={ this.placeholder(placeholder, value) }
-          ref={ this.bridge }
-          value={ this.value(value) }
-          leftChildren={ this.isMultiValue(value) && this.renderMultiValues(value) }
-          { ...events }
+          { ...this.dataAttributes() }
+          { ...this.textboxProps() }
+          { ...this.eventProps() }
+          role='combobox'
+          aria-haspopup='listbox'
+          aria-expanded={ open }
         >
-          { this.state.open && (
+          { this.listDisplayable(allowTypeAhead, filter, open) && (
             <SelectList
-              alwaysHighlight={ !!this.state.filter } // only always ensure something is highlighted if filter
+              alwaysHighlight={ !!filter } // always ensure something is highlighted only if there's a filter
               customFilter={ customFilter }
-              filterValue={ this.state.filter }
+              filterValue={ filter }
               onLazyLoad={ onLazyLoad }
               onMouseEnter={ this.handleMouseEnter }
               onMouseLeave={ this.handleMouseLeave }
               onSelect={ this.handleChange }
-              open={ this.state.open }
+              open={ open }
               target={ this.input.current && this.input.current.parentElement }
+              role='listbox'
             >
               { children }
             </SelectList>
           ) }
-        </InputDecoratorBridge>
+        </Textbox>
       </>
     );
   }
 }
+
+const optionShape = PropTypes.shape({
+  value: PropTypes.string.isRequired,
+  text: PropTypes.string.isRequired
+});
+
+Select.propTypes = {
```

Comment:
Would you like to add description to the props?

Reply:
Added in 2894e25 :thumbsup: